src/libs/OptionsListUtils.ts

```diff
@@ -16,47 +24,97 @@ import * as LoginUtils from './LoginUtils';
 import Navigation from './Navigation/Navigation';
 import Permissions from './Permissions';
 import * as PersonalDetailsUtils from './PersonalDetailsUtils';
+import {PersonalDetailsList} from './PolicyUtils';
 import * as ReportActionUtils from './ReportActionsUtils';
 import * as ReportUtils from './ReportUtils';
 import * as TaskUtils from './TaskUtils';
 import * as TransactionUtils from './TransactionUtils';
 import * as UserUtils from './UserUtils';
 
+type Tag = {enabled: boolean; name: string; accountID: number | null};
+
+type Option = {text: string; keyForList: string; searchText: string; tooltipText: string; isDisabled: boolean};
+
+type PayeePersonalDetails = {text: string; alternateText: string; icons: OnyxCommon.Icon[]; descriptiveText: string; login: string; accountID: number};
+
+type CategorySection = {title: string | undefined; shouldShow: boolean; indexOffset: number; data: Option[] | Participant[] | ReportUtils.OptionData[]};
+
+type Category = {
+    name: string;
+    enabled: boolean;
+};
+
+type Hierarchy = Record<string, Category & {[key: string]: Hierarchy & Category}>;
```

Comment:
I believe this can be rewritten like this:

```suggestion
type Hierarchy = Record<string, Category & Hierarchy>;
```

Reply:
in this way its creating a type error :/

Reply:
@kubabutkiewicz I'm curious about this, how is the data structure of this type?

Reply:
@fabioh8010 
```
{
     Parent: {
       name: "Parent",
       Child: {
          name: "Child"
          Subcategory: {
              name: "Subcategory"
            }
          }
        }
}
```
Its something like that